web/src/App.tsx

```diff
@@ -11,22 +13,32 @@ import { ViewPost } from './pages/view-post';
 import { ROUTES } from './routes';
 import { isDev } from './util';
 
+type UserContext = {
+  currentUser: GetCurrentUserResponse;
+  setCurrentUserGlobally: Dispatch<SetStateAction<GetCurrentUserResponse>>;
```

Comment:
these should probably be optional?
```
{
  currentUser?: ...
  setCurrentUserGlobally?: ...
}
```
then you could use the type `UserContext` when creating the state below: `useState<UserContext>({});`.

Reply:
This didn't work for me as I expect.